backend/prisma/schema.prisma

```diff
@@ -0,0 +1,80 @@
+}
+
+enum report_status {
+  SEEN
+  UNSEEN
+  REMOVED
+  SETTLED
+}
+
+model courses {
+  courseCode         String     @id(map: "pk_course_code") @map("course_code")
+  archived           Boolean
+  attributes         String[]
+  calendar           String
+  campus             String
+  description        String
+  enrolmentRules     String     @map("enrolment_rules")
+  equivalents        String[]
+  exclusions         String[]
+  faculty            String
+  fieldOfEducation   String     @map("field_of_education")
+  genEd              Boolean     @map("gen_ed")
+  level              Int
+  school             String
+  studyLevel         String     @map("study_level")
+  terms              Int[]
+  title              String
+  uoc                Int
+  rating             Float
+  reviews            reviews[]
+}
+
+model reports {
+  reportId          String                      @id(map: "pk_report_id") @default(dbgenerated("gen_random_uuid()")) @db.Uuid @map("report_id")
+  reviewId          String                      @db.Uuid @map("review_id")
+  zid               String
+  status            report_status
+  reason            String
+  createdTimestamp  DateTime                    @default(now()) @db.Timestamp(6) @map("created_timestamp")
+  updatedTimestamp  DateTime                    @default(now()) @db.Timestamp(6) @map("updated_timestamp")
+  reviews           reviews                     @relation(fields: [reviewId], references: [reviewId], onDelete: NoAction, onUpdate: NoAction, map: "fk_review_id")
+  users             users                       @relation(fields: [zid], references: [zid], onDelete: NoAction, onUpdate: NoAction, map: "fk_zid")
+}
+
+model reviews {
+  reviewId          String      @id(map: "pk_review_id") @default(dbgenerated("gen_random_uuid()")) @db.Uuid @map("review_id")
+  zid               String
+  courseCode        String      @map("course_code")
+  authorName        String      @map("author_name")
+  title             String
+  description       String?
+  grade             Int?
+  termTaken        String      @map("term_taken")
+  createdTimestamp  DateTime    @default(now()) @db.Timestamp(6) @map("created_timestamp")
+  updatedTimestamp  DateTime    @default(now()) @db.Timestamp(6) @map("updated_timestamp")
+  upvotes           String[]
+  manageability     Float
+  usefulness        Float
+  enjoyability      Float
+  overallRating     Float       @map("overall_rating")
+  reportes           reports[]
+  courses           courses     @relation(fields: [courseCode], references: [courseCode], onDelete: NoAction, onUpdate: NoAction, map: "fk_course_code")
+  users             users       @relation(fields: [zid], references: [zid], onDelete: NoAction, onUpdate: NoAction, map: "fk_zid")
+}
+
+model users {
+  zid                String      @id(map: "pk_zid")
+  bookmarkedReviews String[]     @map("bookmarked_reviews")
+  bookmarkedCourses String[]     @map("bookmarked_courses")
```

Comment:
unfortunate design decision. These need to be changed in the future to actually be the reviews and courses.